### PR TITLE
show non landscape logos bigger on the login page

### PR DESCRIPTION
**Pull request by icewind1991**
If a custom logo is set, we add a transparent image inside the div to scale the div to the aspect ratio of the logo.

Instead of always limiting the height to `120px` the height is now dynamically scaled between `120px` and `300px` based on the aspect ratio of the logo.

Before:

before

After:

after

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @nextcloud/designers

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

File: core/templates/layout.guest.php [hunk 1]
```diff
@@ -40,6 +40,9 @@
 								<h1 class="hidden-visually">
 									<?php p($theme->getName()); ?>
 								</h1>
+								<?php if(\OC::$server->getConfig()->getAppValue('theming', 'logoMime', false)): ?>
+									<img src="<?php p(\OC::$server->getURLGenerator()->getAbsoluteURL('apps/theming/logo') . '?' . \OC::$server->getConfig()->getAppValue('theming', 'cachebuster', '0')) ?>"/>
```
Comment (MorrisJobke) [inline core/templates/layout.guest.php]: But maybe use:

```
\OC::$server->query(\OCP\Defaults::class)->getLogo()
```

to fetch the logo. This already contains the cache buster and with this approach we also don't need the background image anymore.

**Review by jancborchardt — DISMISSED** [review]
I would reduce the maximum from 300px to 200px for example. Because this is also so it works nicely on mobile, and on your screenshot the logo also simply looks way too big compared to the input fields.

File: core/templates/layout.guest.php (hunk 1 above)
Comment (icewind1991) [inline core/templates/layout.guest.php]: Switched to `getLogo`

I would prefer to keep it as background image for the moment to keep changes minimal

**Comment by icewind1991** [comment]
@jancborchardt lowered max size to 200px

**Comment by MorrisJobke** [comment]
I fixed a little bit the code, because the `$theme` variable could be used directly instead of fetching it via our DI container.

**Comment by Espina2** [comment]
It would be nice if the container is vertically in the center instead of having a margin on top.

@jancborchardt @MorrisJobke

**Review by LukasReschke — APPROVED** [review]
While not perfect still an improvement => Let's merge this for now :)

**Pull request merged**